test-harness/helpers.ts

```diff
@@ -1,3 +1,29 @@
+import { Dictionary } from '@stoplight/types/dist';
+import * as xmlDiff from 'diff-js-xml';
+import * as parser from 'fast-xml-parser';
+import { get, omit } from 'lodash';
+import * as typeIs from 'type-is';
+
+type Result = { body: string; headers: Dictionary<string, string> };
+
+const xmlValidator = {
+  test: (contentType: string, content: string) => {
+    const doesContentTypeMatch = !!typeIs.is(contentType, ['application/xml', 'application/*+xml', 'text/xml']);
+    const isContentXML = parser.validate(content) === true;
+
+    return doesContentTypeMatch || isContentXML;
+  },
+  validate: (expected: Result, output: Result) => {
+    xmlDiff.diffAsXml(expected.body, output.body, {}, { compareElementValues: false }, result =>
+      expect(result).toStrictEqual([]),
+    );
+
+    expect(omit(output, 'body')).toMatchObject(omit(expected, 'body'));
```

Comment:
```suggestion
```

see comment below. this is the same for every validator so could be extracted. up to you